### PR TITLE
add prow test config dumper

### PR DESCRIPTION
**Pull request by tuminoid**
Add prow_test_dumper.py to hack/. It will read the prow config.yaml and spit out branch protection settings it finds. Supports "branch-protection" and "presubmits" sections. Does not support split prow config, so reads only one file at a time.

Requires python3 and pyyaml (`pip3 install pyyaml`). Use virtualenv's responsibly.

Example output: https://github.com/metal3-io/project-infra/pull/714#issuecomment-2078910887 

Script location and naming suggestion welcome.

### CODE REVIEW

File: hack/prow_test_dumper.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+from argparse import Namespace, ArgumentParser
+from typing import Dict
+
+# pip3 install PyYAML
+import yaml
+
+
+def get_external_branchprotection(data: dict) -> Dict[str, list]:
+    """print external branch protection jobs"""
+    protected_repos: Dict[str, list] = {}
```
Comment (NymanRobin) [inline hack/prow_test_dumper.py:15]: This could be defaultdict and would make this function a bit less nested?

File: hack/prow_test_dumper.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    protected_repos[key] = required_checks
+                else:
+                    protected_repos[key].append(required_checks)
+
+    return protected_repos
+
+
+def get_prow_branchprotection(data: dict) -> Dict[str, list]:
+    """get prow required tests"""
+    protected_repos: Dict[str, list] = {}
```
Comment (NymanRobin) [inline hack/prow_test_dumper.py:32]: Same here

**Review by NymanRobin — COMMENTED** [review]
Small nit, probably not crucial for this script :)

File: hack/prow_test_dumper.py (hunk 2 above)
Comment (tuminoid) [reply hack/prow_test_dumper.py]: Cleaned this up in many ways, including defaultdict. Thanks.

**Comment by tuminoid** [comment]
> Small nit, probably not crucial for this script :)

Yeah its quick and dirty implementation, but I wanted to get some comments first before polishing. I cleaned it up now anyways, thanks!

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by Rozzii — APPROVED** [review]
/lgtm

**Pull request merged**